components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -1941,12 +1982,24 @@ def template(m):
                 }
 """
 
-        return self.generateGuardedArray(
-            array, name,
-            template,
-            '    JSPropertySpec::ZERO',
-            'JSPropertySpec',
-            condition, specData)
+        if self.crossorigin:
+            groups = groupby(array, lambda m: condition(m, self.descriptor))
+            assert len(list(groups)) == 1  # can't handle mixed condition
+            elems = [template(m) % specData(m) for m in array]
+            return dedent(
+                """
+                const %s: &[JSPropertySpec] = &[
+                %s,
+                    JSPropertySpec::ZERO,
+                ];
+                """) % (name, ',\n'.join(elems))
+        else:
+            return self.generateGuardedArray(
+                array, name,
+                template,
+                '    JSPropertySpec::ZERO',
+                'JSPropertySpec',
+                condition, specData)
```

Comment:
Do you mean adding a new method similar to `PropertyDefiner.generateGuardedArray`? If that's the case, `generateFlatArray` or `generateUnguardedArray` might be a better name since anything in `PropertyDefiner` obviously pertains to `JSFunctionSpec` or `JSPropertySpec`.

Reply:
Yeah, wherever makes sense. I just saw very similar code being added for JSPropertySpec and JSFunctionSpec arrays, and it would be nice to avoid duplicating that logic.

Reply:
Addressed in 2e0dd0816f8385fda04e69908916e0fe00fb6d39